File: Algorytmy/Python/wyszukiwanie_binarne.py
# Funkcja zwraca indeks szukanego elementu na liście lub None jeśli element nie istnieje
def binary_search(tablica, szukana):
  start = 0
  koniec = len(tablica) - 1

  while start <= koniec:
    mid = (start + koniec) // 2

    if tablica[mid] == szukana:
      return mid
    else:
      if tablica[mid] > szukana:
        koniec = mid -1
      else:
        start = mid + 1

  return None 


tablica = [i for i in range(1,101)] # utworzenie listy z rosnącymi wartościami od 1 do 100
print(binary_search(tablica, 40)) # zwróci 6
print(binary_search(tablica, 59)) # zwróci None

File: Algorytmy/C++/wyszukiwanie_binarne.cpp
#include <iostream>

using namespace std;

// Funkcja zwraca indeks szukanego elementu w tablicy, lub -1, jeśli element nie istnieje
int binary_search(int T[], int S, int L) {

  int start = 0,
      koniec = L - 1;

  while (start <= koniec) {
    int mid = (start + koniec) / 2;

    if (T[mid] == S) {
      return mid;
    }
    else {
      if (mid > S) {
        koniec = mid - 1;
      }
      else {
        start = mid + 1;
      }
    }
  }

  return -1;
}

int main()
{
  int tablica[] = {1,3,4,8,9,10,16,17,20,24,29,30};
  int szukana = 4;
  int len = sizeof(tablica) / sizeof(tablica[0]);

  cout << binary_search(tablica, szukana, len); // zwróci 2

  return 0;
}